File: internal/core/domain/article.go
package domain

import "time"

type Article struct {
	ID          int       `json:"art_id"`
	Slug        string    `json:"slug"`
	Title       string    `json:"title"`
	Excerpt     string    `json:"excerpt"`
	Category    string    `json:"category"`
	ImageURL    string    `json:"image_url"`
	PublishedAt time.Time `json:"published_at"`
	IsPublished bool      `json:"is_published"`
	Content     string    `json:"content"`
	TitleShort  string    `json:"title_short"`
}

File: mobile_app/lib/models/article.dart
class Article {
  final int id;
  final String title;
  final String titleShort;
  final String slug;
  final String category;
  final String excerpt;
  final String imageUrl;
  final String content;
  final int pinOrder;
  final DateTime? publishedAt;

  Article({
    required this.id,
    required this.title,
    required this.titleShort,
    required this.slug,
    required this.category,
    required this.excerpt,
    required this.imageUrl,
    required this.content,
    required this.pinOrder,
    this.publishedAt,
  });

  factory Article.fromJson(Map<String, dynamic> json) {
    return Article(
      id: json['art_id'] ?? json['message_id'] ?? 0, // Fallback support
      title: json['title'] ?? '',
      titleShort: json['title_short'] ?? '',
      slug: json['slug'] ?? '',
      category: json['category'] ?? '',
      excerpt: json['excerpt'] ?? '',
      imageUrl: json['image_url'] ?? '',
      content: json['content'] ?? '',
      pinOrder: json['pin_order'] ?? 0,
      publishedAt: json['published_at'] != null 
          ? DateTime.tryParse(json['published_at']) 
          : null,
    );
  }
}
